### PR TITLE
PLAT-683 - Remove postdeploy script and add data migrations

### PR DESCRIPTION
**Pull request by danielsousaio**

### CODE REVIEW

File: {{cookiecutter.project_slug}}/migrations.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    HomePage.objects.create(body=homepage_body)
+
+
+def create_site(apps, schema_editor):
+    site = Site.objects.first()
+    if site:
+        site.name = "{{cookiecutter.project_name}}"
+        site.domain = "{{cookiecutter.custom_domain}}" or site.domain
+        site.save()
```
Comment (danielsousaio) [inline {{cookiecutter.project_slug}}/migrations.sh]: This needs another solution, the table isn't created at this moment yet.

Comment (danielsousaio) [reply {{cookiecutter.project_slug}}/migrations.sh]: Fixed in 8959050

**Comment by danielsousaio** [comment]
> Why is this done in a script rather than having actual migrations committed to the repo?

I can't add the `0002` migration because the scaffold doesn't hold the initial migrations. If we add it, then `makemigrations` will fail because the migration `0001` isn't created yet.

We could obviously add the migrations directly to the scaffold, with 2 **very very minor** disadvantages at first glance:
- Adding new migrations to the scaffold isn't a command away, due to the cookiecutter placeholder variables.
- The migrations timestamps would not be fresh.

I just continued our current structure, which is to generate the migrations in the scaffold process by running `makemigrations` there.

**Comment by danielsousaio** [comment]
@driverdan As we discussed directly, the two commits add the migrations on this repo. This also meant removing the `makemigrations` step in the Django scaffold process in our codebase.

Respective `crowdbotics-slack-app` commit:
https://github.com/crowdbotics/crowdbotics-slack-app/pull/1415/commits/12dfdb26647cfb93d20e2ec8c1a73a27c340cfeb

File: {{cookiecutter.project_slug}}/home/migrations/0002_load_initial_data.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    HomePage.objects.create(body=homepage_body)
+
+
+def create_site(apps, schema_editor):
+    Site = apps.get_model("sites", "Site")
+
+    site_params = {
+        "name": "{{cookiecutter.project_name}}",
+        "domain": "{{cookiecutter.custom_domain}}",
```
Comment (driverdan) [inline {{cookiecutter.project_slug}}/home/migrations/0002_load_initial_data.py]: `custom_domain` could be empty. It shouldn't be changed in that case.

Comment (danielsousaio) [reply {{cookiecutter.project_slug}}/home/migrations/0002_load_initial_data.py]: Fixed in b63fa02

File: {{cookiecutter.project_slug}}/home/migrations/0002_load_initial_data.py [hunk 3]
```diff
@@ -22,11 +22,13 @@ def create_homepage(apps, schema_editor):
...
 def create_site(apps, schema_editor):
     Site = apps.get_model("sites", "Site")
+    custom_domain = "{{cookiecutter.custom_domain}}"
 
     site_params = {
         "name": "{{cookiecutter.project_name}}",
-        "domain": "{{cookiecutter.custom_domain}}",
     }
+    if custom_domain:
+        site_params.update({"domain": custom_domain})
```
Comment (driverdan) [inline {{cookiecutter.project_slug}}/home/migrations/0002_load_initial_data.py]: This can be simplified to `site_params["domain"] = custom_domain`

Comment (danielsousaio) [reply {{cookiecutter.project_slug}}/home/migrations/0002_load_initial_data.py]: True but it comes down to personal preference, I guess. In this instance we are adding only a key value pair so I'll make the change.

**Pull request merged**